### DISCUSSION TITLE
configuring which websites can make API request

### DISCUSSION THREAD

**Discussion by dovenokan** [supabase/supabase#12327] [Feature Requests]
# Feature request

## Is your feature request related to a problem? Please describe.

I have "website X" and I am using supabase API. 
If I'm not using middleware, someone steal my api from source code easily.

## Describe the solution you'd like

if request_url == "website X" => then allow request
So I don't need a middleware for my small projects.

## Additional context

If I'm not aware of what I suggest or supabase has something like that, please tell me. 
Thanks!

**Comment by kiwicopple** [comment]
This should be doable today, using something like pg_headerkit: https://github.com/supabase-community/pg_headerkit

You can add these to your RLS policies
